casanovo/data/db_utils.py

```
@@ -234,21 +244,24 @@ def get_candidates(
-        return self.db_peptides.index[mask]
+        neutral_mass = float(_to_neutral_mass(precursor_mz, charge))
+        isotope_offsets = np.arange(
+            self.isotope_error[0], self.isotope_error[1] + 1
+        )
+        shifted_masses = neutral_mass - ISOTOPE_SPACING * isotope_offsets
+        lower_bounds = shifted_masses * (1 - precursor_tol_ppm)
+        upper_bounds = shifted_masses * (1 + precursor_tol_ppm)
+        los = np.searchsorted(masses, lower_bounds, side="left")
+        his = np.searchsorted(masses, upper_bounds, side="right")
+        positions = [np.array([], dtype=np.intp)]
+        for lo, hi in zip(los, his):
```

Ruff (0.15.7)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)